lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1027,18 +1032,36 @@ def view_init(self, elev=None, azim=None, roll=None, vertical_axis="z"):
             dict(x=0, y=1, z=2), vertical_axis=vertical_axis
         )
 
-    def set_proj_type(self, proj_type):
+    def set_proj_type(self, proj_type, focal_length=None):
         """
         Set the projection type.
 
         Parameters
         ----------
         proj_type : {'persp', 'ortho'}
-        """
-        self._projection = _api.check_getitem({
-            'persp': proj3d.persp_transformation,
-            'ortho': proj3d.ortho_transformation,
-        }, proj_type=proj_type)
+            The projection type.
+        focal_length : float, default: None
+            For a projection type of 'persp', the focal length of the virtual
+            camera. Must be > 0. If None, defaults to 1.
+            The focal length can be computed from a desired Field Of View via
+            the equation: focal_length = 1/tan(FOV/2)
+        """
+        if proj_type == 'persp':
+            if focal_length is None:
+                self.focal_length = 1
```

Comment:
Does `focal_length` need to be a public attribute? It should be set through `set_proj_type` to go through verification.

Reply:
Up to you guys if you want the leading underscore, I'm not super familiar with the conventions here.

Reply:
If there is an underscore it is "private" and users only manipulate it at their peril.  If we make it public, we have to make it do something sensible, and we can't remove it w/o deprecation.

Reply:
Copy that, added the underscore